X86/X86MachineInstructionRaiser.cpp

```diff
@@ -99,7 +99,8 @@ bool X86MachineInstructionRaiser::raisePushInstruction(const MachineInstr &MI) {
   // If the value of push operand register is known, then it does not
   // have a manifestation in the function code. So, a known garbage value is
   // used to affect the push operation.
-  Type *StackRefElemTy = StackRef->getType()->getPointerElementType();
```

Comment:
Please consider definition of the following helper function based on information [here](https://llvm.org/docs/OpaquePointers.html#migration-instructions). The intent is to use this as replacement of the deprecated upstream API, as much as possible in a cleaner manner keeping away all the unwrapping to one helper function call.

For example, we'd use it here as ` Type *StackRefElemTy = getPointerElementType(StackRef);`

```
// Helper function added since LLVM deprecated getPointerElementType() API
// Get type of a pointer or non-pointer typed Val
Type *X86MachineInstructionRaiser::getPointerElementType(const Value *Val) {
  Type *RetTy = nullptr;
  if (Val->getType()->isPointerTy()) {
    if (isa<AllocaInst>(Val)) {
      RetTy = dyn_cast<AllocaInst>(Val)->getAllocatedType();
    } else if (isa<GlobalVariable>(Val)) {
      RetTy = dyn_cast<GlobalVariable>(Val)->getValueType();
    } else if (isa<GetElementPtrInst>(Val)) {
      RetTy = dyn_cast<GetElementPtrInst>(Val)->getSourceElementType();
    } else if (Val->getType()->isPointerTy() && isa<GEPOperator>(Val)) {
      RetTy = dyn_cast<GEPOperator>(Val)->getSourceElementType();
    } else if (isa<LoadInst>(Val)) {
      RetTy = dyn_cast<LoadInst>(Val)->getType();
    } else if (isa<StoreInst>(Val)) {
      RetTy = dyn_cast<StoreInst>(Val)->getValueOperand()->getType();
    } else {
      assert(RetTy != nullptr && "Unhandled pointer type");
    }
  }
  return RetTy;
}
```